### ISSUE TITLE
Feature Request: Consider returning something other than a link to empty list when pagination is exhausted

### ISSUE THREAD

**Issue by ire-and-curses** [stellar/go#3399]
### What problem does your feature solve?
Users are often [surprised](https://github.com/stellar/js-stellar-sdk/issues/600) to discover that Horizon continues to return `next` and `prev` links, even when data is exhausted. This response continues indefinitely. The current expectation is for client to check for a zero length list of records and stop looping.

### What would you like to see?

We should consider whether there is a more intuitive or standard way to handle reaching the end of available data, when we overhaul the API.

### What alternatives are there?

We can leave it the way it is (and maybe improve docs). It works, and each client only makes this mistake once.

**Comment by leighmcculloch** [comment]
>We can leave it the way it is (and maybe improve docs). It works, and each client only makes this mistake once.

I don't think this alternative solves all the problems created by this feature of the API.

It requires SDKs and clients to make an additional request, or to write error prone logic to avoid it. From my own experience writing a tool that collects effects on transactions, this means 2x the number of HTTP requests because every collection of a list requires at least two requests, four technically for web based clients:
 - OPTIONS for CORS
 - GET for the data
 - OPTIONS for CORS
 - GET for the empty page of data

For clients to avoid ☝🏻 they have to manually specify the page size, and then check that the results they got back are equal to the page size or not. If they are equal, get the next page just in case there is more, if there is less then they can assume there are no more.

For a client that looks up effects for 5000 transactions, that's 5000 extra requests, 10000 on web.

@ire-and-curses I saw you leave this comment on :

>For the next iteration of the API this is something we can consider, if there's a better approach

I realize we probably can't remove the `next` link because that would be a breaking change, but we could include some anciliary information. Some ideas:

- Include a `has_more` boolean alongside the next URL, set to true 
- Include a total count so the client can compare it with their page size

**Comment by ire-and-curses** [comment]
Yep I agree. FWIW the doubling is amortized for lists with multiple pages of data though.

I like the suggestions!

**Comment by leighmcculloch** [comment]
> the doubling is amortized for lists with multiple pages of data though.

This depends on the endpoint. If effects are pulled via their transaction or operation, most of the time the dataset returned will be less than 200, the maximum page size, resulting in one page of data, and an extra request. A system that iterates over transactions pulling their effects will download 2x the number of requests.

I agree for endpoints that typically return many pages this matters little.
